vizier/backend/src/info/vizierdb/catalog/Artifact.scala

```diff
@@ -248,6 +249,26 @@ case class Artifact(
                  Map(id -> this), 
                  Artifact.get(_:Identifier)
                )
+
+    
+    val df = dataframe(session)()
+    //println(datasetDescriptor.properties)
+    //println(datasetDescriptor.properties.get("is_profiled"))
+    
+    // if the profiler is false, then run the profiler and turn it into true
+    // Check if "is_profiled" is present and set it to true
+    val updatedProperties = datasetDescriptor.properties.get("is_profiled") match {
+      case Some(jsValue) if jsValue.isInstanceOf[JsObject] =>
+        // Profiled information is already attached, do nothing.
+
+      case _ =>
+        // add individual properties, use a field to update each one at a time.
+        val dataProfile: Map[String, JsValue] = DataProfiler.apply(df)
```

Comment:
The profile is being generated, but it doesn't look like it's getting saved anywhere.  `updateDatasetProperty("is_profiled", json)`: The `json` there is just getting a record of the data descriptor`